pulp_ostree/app/tasks/uploading.py

```diff
@@ -0,0 +1,263 @@
+        _, related_objects = self.repo.traverse_commit(commit.checksum, maxdepth=0)
+        for obj_checksum, obj_type in related_objects.values():
+            if obj_checksum == commit.checksum:
+                continue
+
+            obj = OstreeObject(commit=commit, typ=obj_type, checksum=obj_checksum)
+
+            # OSTree objects, such as, file and filez objects are described by the same object type;
+            # the ostree-libs API does not provide methods for retrieving the filename of an object;
+            # therefore, we determine a correct filepath by utilizing the glob.glob() function and
+            # UNIX wildcards
+            _, filename = os.path.split(
+                glob.glob(
+                    os.path.join(self.repo_path, get_checksum_filepath(obj_checksum, obj_type))
+                )[0]
+            )
+            obj_relative_path = os.path.join("objects/", obj_checksum[:2], filename)
+
+            object_dc = self.create_artifact_dc(obj_relative_path, obj)
+            await self.put(object_dc)
+
+
+class OstreeUploadDeclarativeVersion(DeclarativeVersion):
+    """A customized DeclarativeVersion class that creates a pipeline for the OSTree uploads."""
+
+    def pipeline_stages(self, new_version):
+        """Build a list of stages."""
+        pipeline = [
+            self.first_stage,
+            QueryExistingArtifacts(),
+            ArtifactSaver(),
+            QueryExistingContents(),
+            ContentSaver(),
+            OstreeAssociateContent(),
+        ]
+
+        return pipeline
+
+
+class OstreeAssociateContent(Stage):
+    """A stage for creating associations between OSTree objects."""
+
+    async def run(self):
+        """Create relations between each OSTree object specified in DeclarativeContent objects."""
+        async for batch in self.batches():
+            updated_commits = []
+            for dc in batch:
+                if dc.extra_data.get("parent_commit"):
+                    updated_commits.append(self.associate_parent_commit(dc))
+
+            OstreeCommit.objects.bulk_update(
+                objs=updated_commits, fields=["parent_commit"], batch_size=1000
+            )
+            for dc in batch:
+                await self.put(dc)
+
+    def associate_parent_commit(self, dc):
+        """Assign the parent commit to its child commit."""
+        parent_commit_dc = dc.extra_data.get("parent_commit")
+        dc.content.parent_commit = parent_commit_dc.content
+        return dc.content
+
+
+def get_checksum_filepath(checksum, obj_type):
+    """Return an object's relative filepath within a repository based on its checksum and type."""
+    extension = get_file_extension(obj_type)
+    return os.path.join("objects/", checksum[:2], f"{checksum[2:]}.{extension}")
+
+
+def get_file_extension(obj_type):
+    """Return a file extension based on the type of the object."""
+    if obj_type == OstreeObjectType.OSTREE_OBJECT_TYPE_FILE:
+        return "file*"
```

Comment:
is this 
```suggestion
        return "file"
```
?

Reply:
No, because it can be either "file" or "filez". Using wildcards helps us to determine the correct file extension.